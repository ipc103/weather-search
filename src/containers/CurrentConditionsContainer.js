import React, { Component } from 'react'

import { fetchCurrentConditions } from '../api/weatherUnderground'
import SearchForm from '../components/SearchForm'
import CurrentConditionsDetail from '../components/CurrentConditionsDetail'

class CurrentConditionsContainer extends Component {
  constructor(){
    super()
    this.state = {
      city: '',
      st: '',
      conditions: {
        message: 'Enter a city and state abbreviation'
      }
    }
    this.handleQueryChange = this.handleQueryChange.bind(this)
    this.handleSearch      = this.handleSearch.bind(this)
  }

  handleQueryChange(event){
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  handleSearch(event){
    event.preventDefault()
    fetchCurrentConditions(this.state.city, this.state.st)
      .then(data => {
        // TODO: clean this up
        if (data.current_observation) {
          this.setState({
            city: '',
            st: '',
            conditions: {
              temp: data.current_observation.temperature_string,
              description: data.current_observation.weather,
              city: `${data.location.city}, ${data.location.state}`
            }
          })
        } else {
          this.setState({
            conditions: {
              message: 'No results found'
            }
          })
        }
      })
  }

  render(){
    return(
      <div className='container'>
        <div className='col-md-4'>
          <SearchForm city={this.state.city} state={this.state.st} handleChange={this.handleQueryChange} handleSubmit={this.handleSearch} />
        </div>
        <div className='col-md-8'>
          <CurrentConditionsDetail conditions={this.state.conditions} />
        </div>
      </div>
    )
  }
}

export default CurrentConditionsContainer
