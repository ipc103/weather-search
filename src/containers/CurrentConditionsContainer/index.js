import React, { Component } from 'react'

import { fetchCurrentConditions } from '../../api/weatherUnderground'
import SearchForm from '../../components/SearchForm'
import CurrentConditionsDetail from '../../components/CurrentConditionsDetail'

class CurrentConditionsContainer extends Component {
  constructor(){
    super()
    this.state = {
      city: '',
      region: '',
      conditions: {
        message: 'Enter a city and state abbreviation'
      }
    }
    this.handleQueryChange = this.handleQueryChange.bind(this)
    this.handleSearch      = this.handleSearch.bind(this)
    this.parseWeather      = this.parseWeather.bind(this)
  }

  handleQueryChange(event){
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  handleSearch(event){
    event.preventDefault()
    fetchCurrentConditions(this.state.city, this.state.region)
      .then(this.parseWeather)
  }

  parseWeather(data){
    if (data.current_observation) {
      this.setState({
        city: '',
        region: '',
        conditions: {
          temp: data.current_observation.temperature_string,
          weather: data.current_observation.weather.toLowerCase(),
          city: `${data.location.city}, ${data.location.state}`
        }
      })
    } else {
      this.setState({
        conditions: {
          message: 'No results found'
        }
      })
    }
  }

  render(){
    return(
      <div className='container'>
        <div className='col-md-4'>
          <SearchForm city={this.state.city} region={this.state.region} handleChange={this.handleQueryChange} handleSubmit={this.handleSearch} />
        </div>
        <div className='col-md-8'>
          <CurrentConditionsDetail conditions={this.state.conditions} />
        </div>
      </div>
    )
  }
}

export default CurrentConditionsContainer
