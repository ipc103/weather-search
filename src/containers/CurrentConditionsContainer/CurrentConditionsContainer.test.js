import React from 'react'
import {shallow, mount} from 'enzyme'
import CurrentConditionsContainer from './'

jest.mock('../../api/weatherUnderground')

describe('CurrentConditionsContainer', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<CurrentConditionsContainer/>)
    expect(wrapper).toMatchSnapshot()
  })

  describe('searching for a city and region', () => {
    const wrapper = mount(<CurrentConditionsContainer/>)

    describe('with valid city and region', () => {
      beforeEach(() => {
        wrapper.setState({city: 'Pittsburgh', region: 'PA'})
        const event = new Event('submit')
        wrapper.instance().handleSearch(event)
      })

      it('updates the state with current conditions',() => {
        const expected = {
          city: '',
          region: '',
          conditions: {
            weather: 'partly cloudy',
            temp: '76.6 F (24.8 C)',
            city: `Pittsburgh, PA`
          }
        }
        expect(wrapper.state()).toEqual(expected)
      })
    })
  })
})
