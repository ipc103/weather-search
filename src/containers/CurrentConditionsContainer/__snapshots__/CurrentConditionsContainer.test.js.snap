// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentConditionsContainer renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="container"
>
    <div
        className="col-md-4"
    >
        <Unknown
            city=""
            handleChange={[Function]}
            handleSubmit={[Function]}
            region=""
        />
    </div>
    <div
        className="col-md-8"
    >
        <Unknown
            conditions={
                Object {
                    "message": "Enter a city and state abbreviation",
                  }
            }
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="container"
>
      <div
            className="col-md-4"
      >
            <Unknown
                  city=""
                  handleChange={[Function]}
                  handleSubmit={[Function]}
                  region=""
            />
      </div>
      <div
            className="col-md-8"
      >
            <Unknown
                  conditions={
                        Object {
                              "message": "Enter a city and state abbreviation",
                            }
                  }
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CurrentConditionsContainer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CurrentConditionsContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleQueryChange": [Function],
        "handleSearch": [Function],
        "parseWeather": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "city": "",
          "conditions": Object {
            "message": "Enter a city and state abbreviation",
          },
          "region": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="container"
>
          <div
                    className="col-md-4"
          >
                    <Unknown
                              city=""
                              handleChange={[Function]}
                              handleSubmit={[Function]}
                              region=""
                    />
          </div>
          <div
                    className="col-md-8"
          >
                    <Unknown
                              conditions={
                                        Object {
                                                  "message": "Enter a city and state abbreviation",
                                                }
                              }
                    />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="container"
>
          <div
                    className="col-md-4"
          >
                    <Unknown
                              city=""
                              handleChange={[Function]}
                              handleSubmit={[Function]}
                              region=""
                    />
          </div>
          <div
                    className="col-md-8"
          >
                    <Unknown
                              conditions={
                                        Object {
                                                  "message": "Enter a city and state abbreviation",
                                                }
                              }
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CurrentConditionsContainer />,
}
`;
